scenario/runtime.py

```diff
@@ -180,25 +180,24 @@ def _cleanup_env(env):
 
     def _get_event_env(self, state: "State", event: "Event", charm_root: Path):
         """Build the simulated environment the operator framework expects."""
-        if event.name.endswith("_action"):
-            # todo: do we need some special metadata, or can we assume action names
-            #  are always dashes?
-            action_name = event.name[: -len("_action")].replace("_", "-")
-        else:
-            action_name = ""
-
         env = {
             "JUJU_VERSION": self._juju_version,
             "JUJU_UNIT_NAME": f"{self._app_name}/{self._unit_id}",
             "_": "./dispatch",
             "JUJU_DISPATCH_PATH": f"hooks/{event.name}",
             "JUJU_MODEL_NAME": state.model.name,
-            "JUJU_ACTION_NAME": action_name,
             "JUJU_MODEL_UUID": state.model.uuid,
             "JUJU_CHARM_DIR": str(charm_root.absolute()),
             # todo consider setting pwd, (python)path
         }
 
+        if event.name.endswith("_action"):
```

Comment:
I reckon this code can be simplified thanks to various helpers I added after I originally wrote this.

I added at some point utility properties to check standard event types.

we can also use `event.action.name` instead of parsing again the event name.

If I were to write this again I'd do:


```python
if event._is_action_event and (action := event.action):
     # it should never happen that there is no action, unless you bypassed the consistency checker.
     
      env.update(
            {
                 "JUJU_ACTION_NAME": action.name,
                 "JUJU_ACTION_UUID": action.id,
             }
         )            
```